### PR TITLE
Feat/v3 support

### PR DESCRIPTION
**Pull request by raiseandfall**
## Feature
- added Blockcerts V3 support
- handle new `display` property for Blockcerts layout rendering

## Refactor
- `CertificateParser`: determine Blockcerts version from `@context`

### CODE REVIEW

File: Frameworks/Blockcerts/CertificateV3.swift [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+            issuerId = issuerJSON?["id"] as! String
+        }
+        
+        guard let issuerData = issuerJSON as? [String : String],
+            let issuerURLString = issuerData["url"],
+            let issuerURL = URL(string: issuerURLString),
+            let logoURI = issuerData["image"],
+            let issuerEmail = issuerData["email"],
+            let issuerName = issuerData["name"],
+            let issuerDescription = issuerData["description"],
```
Comment (Capspam) [inline Frameworks/Blockcerts/CertificateV3.swift]: This variable is not used. Please ignore if it is intensional.

File: Frameworks/Blockcerts/CertificateV3.swift [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        
+        return Assertion(issuedOn: issuedOnDate,
+                         uid: assertionUID,
+                         id: assertionIDURL,
+                         metadata: Metadata(json: metadataJson),
+                         htmlDisplay: htmlDisplay)
+    }
+    
+    static func parse(verifyJSON: [String: AnyObject?]) -> Verify? {
+        guard let verifyData = verifyJSON as? [String : AnyObject],
```
Comment (Capspam) [inline Frameworks/Blockcerts/CertificateV3.swift]: Nit: verifyJSON is already  [String: AnyObject?]. The Casting is redundant.

File: Frameworks/Blockcerts/CertificateV3.swift [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+                        url: issuerURL,
+                        publicKeys: [],
+                        introductionMethod: .unknown,
+                        analyticsURL: nil)
+    }
+    
+    static func parse(recipientJSON json: [String : Any]) -> Recipient? {
+        if let recipientData = json["credentialSubject"] as? [String : Any] {
+            let publicKey = recipientData["publicKey"] as? BlockchainAddress
+            let name : String = recipientData["name"] as? String ?? recipientData["id"] as! String
```
Comment (Capspam) [inline Frameworks/Blockcerts/CertificateV3.swift:198]: Nit: Should this be ```recipientData["id"] as! String``` or ```recipientData["id"] as? String```? Please ignore if you meant this way - ie - let it crash if numbers are passed as id. I am not sure about context though.

File: Frameworks/Blockcerts/Certificate.swift [hunk 4]
```diff
@@ -50,6 +52,43 @@ public enum CertificateParserError : Error {
 
 /// CertificateParser should never be instantiated. Call one of its `parse` methods to turn a Data object into a Certificate.
 public enum CertificateParser {
```
Comment (Capspam) [inline Frameworks/Blockcerts/Certificate.swift]: Is there any reason to make it enum?

File: Frameworks/Blockcerts/CertificateV3.swift (hunk 1 above)
Comment (raiseandfall) [reply Frameworks/Blockcerts/CertificateV3.swift]: 👍

File: Frameworks/Blockcerts/CertificateV3.swift (hunk 2 above)
Comment (raiseandfall) [reply Frameworks/Blockcerts/CertificateV3.swift]: 👍

File: Frameworks/Blockcerts/CertificateV3.swift (hunk 3 above)
Comment (raiseandfall) [reply Frameworks/Blockcerts/CertificateV3.swift:198]: Yeah that was meant to be this way.

File: Frameworks/Blockcerts/Certificate.swift (hunk 4 above)
Comment (raiseandfall) [reply Frameworks/Blockcerts/Certificate.swift]: Hmm I have no idea, this was done by another dev before me (probably better experienced too :))
A `class` would indeed make more sense there.

**Review by Capspam — APPROVED** [review]
Looks good! Thank you!

**Review by Spambit — APPROVED** [review]
Looks good!

**Review by Spambit — APPROVED** [review]

**Pull request merged**